test/concentrate/vehicle_position_test.exs

```diff
@@ -23,5 +58,268 @@ defmodule Concentrate.VehiclePositionTest do
+                %{id: "swiftly", value: nil},
+                %{id: "busloc", value: "busloc_trip"}
+              ]
+            },
+            %DataDiscrepancy{
+              attribute: :route_id,
+              sources: [
+                %{id: "swiftly", value: nil},
+                %{id: "busloc", value: "busloc_route"}
+              ]
+            }
+          ]
+        )
+
+      assert Mergeable.merge(swiftly, non_swiftly) == expected
+      assert Mergeable.merge(non_swiftly, swiftly) == expected
+    end
+
+    test "merge/2 defaults to the latest value if both come from swiftly" do
+      swiftly =
+        new(
+          last_updated: 1,
+          latitude: 1,
+          longitude: 1,
+          trip_id: "swiftly_trip",
+          route_id: "swiftly_route",
+          sources: MapSet.new(["swiftly"])
+        )
+
+      swiftly_merged =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "swiftly_merged_trip",
+          route_id: "swiftly_merged_route",
+          sources: MapSet.new(["busloc", "swiftly"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "swiftly_merged_trip",
+          route_id: "swiftly_merged_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
+            %DataDiscrepancy{
+              attribute: :trip_id,
+              sources: [
+                %{id: "swiftly", value: "swiftly_trip"},
+                %{id: "busloc|swiftly", value: "swiftly_merged_trip"}
+              ]
+            },
+            %DataDiscrepancy{
+              attribute: :route_id,
+              sources: [
+                %{id: "swiftly", value: "swiftly_route"},
+                %{id: "busloc|swiftly", value: "swiftly_merged_route"}
+              ]
+            }
+          ]
+        )
+
+      assert Mergeable.merge(swiftly_merged, swiftly) == expected
+      assert Mergeable.merge(swiftly, swiftly_merged) == expected
+    end
+
+    test "merge/2 doesn't include any data discrepancies if they values are the same" do
+      first =
+        new(
+          last_updated: 1,
+          latitude: 1,
+          longitude: 1,
+          trip_id: "trip",
+          route_id: "route",
+          sources: MapSet.new(["first"])
+        )
+
+      second =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "trip",
+          route_id: "route",
+          sources: MapSet.new(["second"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "trip",
+          route_id: "route",
+          sources: MapSet.new(["first", "second"]),
+          data_discrepancies: []
+        )
+
+      assert Mergeable.merge(first, second) == expected
+    end
+  end
+
+  describe "comes_from_swiftly/1" do
```

Comment:
I thought this was `comes_from_swiftly?/1` (with the question mark) since it returns a boolean?

Reply:
Good call, updated